### ISSUE TITLE
Generated Informers for non controller-runtime use

### ISSUE THREAD

**Issue by youngnick** [kubernetes-sigs/service-apis#61]
Already-written controllers that do not use controller-runtime and kubebuilder will find it difficult to consume these APIs without having generated Informers.

This issue proposes that we add automated generation of the required code to use the new APIs with Informers (so, a clientset, listers, and informers). I'll prepare a PR with the change as well for review.

This will make the implementation of these types in an experimental fashion in Contour and any other controller that is still using Informers much easier.

**Comment by robscott** [comment]
@youngnick We agreed on Services API call today that this makes sense.

**Comment by youngnick** [comment]
So, I spent some time looking into how to do this, and it seems that, because the Group is `networking.x.k8s.io`, the old-style generation tools decide that all the objects are part of the `core` Group instead.

It seems that this is fixable for `client-gen`, as it has ways to override the Group detection, but `lister-gen` and `informer-gen` don't. If anyone has any more experience with the older tools, I'd be happy to accept help, otherwise, I'll head to api-machinery and ask.

It currently looks to me like `lister-gen` and `informer-gen` might need to have similar code to `client-gen` grafted into them for this to work.

In parallel, Contour is looking at moving to the Dynamic client, which is going better than we thought. So I'll timebox this to a couple more days for now, and see if I can push a branch somewhere that has where I've got up to then.
